src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,374 @@
+        {
+            XElement? packageSourcesElement = doc.Root.Element("packageSources");
+            if (packageSourcesElement != null)
+            {
+                XElement? clearElement = packageSourcesElement.Element("clear");
+                if (clearElement != null)
+                {
+                    XElement newAddElement = new XElement("add",
+                    new XAttribute("key", "local-packages"),
+                        new XAttribute("value", localPackagesPath));
+
+                    clearElement.AddAfterSelf(newAddElement);
+                }
+            }
+
+            doc.Save(nuGetConfig);
+        }
+    }
+
+    private void DistroTest(string baseImage, DistroType distroType)
+    {
+        // Order of installation is important as we do not want to use "--nodeps"
+        // We install in correct order, so package dependencies are present.
+
+        // Prepare the package list in correct install order
+        List<string> packageList =
+        [
+            // Deps package should be installed first
+            Path.GetFileName(GetMatchingDepsPackage(baseImage, distroType))
+        ];
+
+        // Add all other packages in correct install order
+        AddPackage(packageList, "dotnet-host-", distroType);
+        AddPackage(packageList, "dotnet-hostfxr-", distroType);
+        AddPackage(packageList, "dotnet-runtime-", distroType);
+        AddPackage(packageList, "dotnet-targeting-pack-", distroType);
+        AddPackage(packageList, "aspnetcore-runtime-", distroType);
+        AddPackage(packageList, "aspnetcore-targeting-pack-", distroType);
+        AddPackage(packageList, "dotnet-apphost-pack-", distroType);
+        if (Config.Architecture == "x64")
+        {
+            // netstandard package exists for x64 only
+            AddPackage(packageList, "netstandard-targeting-pack-", distroType);
+        }
+        AddPackage(packageList, "dotnet-sdk-", distroType);
+
+        string dockerfile = GenerateDockerfile(packageList, baseImage, distroType);
+
+        string tag = $"test-{Path.GetRandomFileName()}";
+        string output = "";
+
+        try
+        {
+            // Build docker image and run the tests
+            _dockerHelper.Build(tag, dockerfile: dockerfile, contextDir: _contextDir);
+            output = _dockerHelper.Run(tag, tag);
+
+            int testResultsSummaryIndex = output.IndexOf("Tests run: ");
+            if (testResultsSummaryIndex >= 0)
+            {
+                string testResultsSummary = output[testResultsSummaryIndex..];
+                Assert.False(AnyTestFailures(testResultsSummary), testResultsSummary);
+            }
+            else
+            {
+                Assert.Fail("Test summary not found");
+            }
+        }
+        catch (Exception e)
+        {
+            if (string.IsNullOrEmpty(output))
+            {
+                output = e.Message;
+            }
+            Assert.Fail($"Build failed: {output}");
```

Comment:
How do you know this failed during the build vs run?

Reply:
We don't - will refactor this.

Reply:
Fixed with https://github.com/dotnet/sdk/commit/64fa1b8e6d800687ab49eae0b2a483e937a83e73